pkg/apis/application/v1alpha1/types.go

```diff
@@ -894,7 +894,9 @@ type ApplicationSourceDirectory struct {
 	// Recurse specifies whether to scan a directory recursively for manifests
 	Recurse bool `json:"recurse,omitempty" protobuf:"bytes,1,opt,name=recurse"`
 	// Jsonnet holds options specific to Jsonnet
-	Jsonnet ApplicationSourceJsonnet `json:"jsonnet,omitempty" protobuf:"bytes,2,opt,name=jsonnet"`
+	// Note: omitzero is used to prevent the field from being marshaled when it is empty.
```

Comment:
Can we drop the two new docs lines? They're helpful for someone reading the code, but they also end up in stuff like swagger docs and kubectl describe output, where they'll probably be more confusing than helpful